lib/types/orderBook.ts

```diff
@@ -8,12 +8,13 @@ export type PlaceOrderResult = {
 };
 
 export type PlaceOrderEvent = {
-  case: PlaceOrderEventCase;
-  payload: OwnOrder | SwapResult;
+  type: PlaceOrderEventType;
+  payload: OwnOrder | SwapResult | PeerOrder;
 };
 
-export enum PlaceOrderEventCase {
+export enum PlaceOrderEventType {
   InternalMatch,
   SwapResult,
   RemainingOrder,
+  FailedSwap,
```

Comment:
`SwapFailure` might be more appropriate as an event name, and that's our current used terminology (`FailedSwap` could work for the payload).
We might also want to change `SwapResult` to `SwapSuccess`, because seeing these two together will make sense.